modules/web/client/src/main/scala/observe/ui/components/MainApp.scala

```diff
@@ -145,80 +188,112 @@ object MainApp:
           AppContext.version(appConfig.environment),
           SSOClient(appConfig.sso),
           (tab: AppTab) => MainApp.routerCtl.urlFor(tab.getPage).value,
-          (tab: AppTab, via: SetRouteVia) => MainApp.routerCtl.set(tab.getPage, via)
+          (tab: AppTab, via: SetRouteVia) => MainApp.routerCtl.set(tab.getPage, via),
+          toastRef
         )
       .useStateView(Pot.pending[RootModelData])
-      .useEffectWithDepsBy((_, ctxPot, _) => ctxPot.void): (_, ctxPot, rootModelData) =>
-        _ => // Once AppContext is ready, proceed to attempt login.
-          ctxPot.toOption
-            .map: ctx =>
-              import ctx.given
+      .useEffectWhenDepsReady((_, _, _, ctxPot, _) => ctxPot): (_, _, _, _, rootModelData) =>
+        ctx => // Once AppContext is ready, proceed to attempt login.
+          import ctx.given
 
-              enforceStaffRole(ctx.ssoClient).attempt
-                .flatMap(userVault =>
-                  rootModelData.async.set(RootModelData.initial(userVault).ready)
-                )
-            .orEmpty
+          enforceStaffRole(ctx.ssoClient).attempt
+            .flatMap(userVault => rootModelData.async.set(RootModelData.initial(userVault).ready))
       .useStateView(Pot.pending[Environment])
       // Subscribe to client event stream (and initialize Environment)
       // TODO Reconnecting middleware
-      .useResourceOnMount(WebSocketClient[IO].connectHighLevel(WSRequest(EventWsUri)))
-      .useAsyncEffectWithDepsBy((_, ctx, rootModelData, environment, wsConnection) =>
-        (wsConnection.void, rootModelData.get.void, ctx.void).tupled
-      ): (_, ctxPot, rootModelDataPot, environment, wsConnectionPot) =>
-        _ =>
-          (wsConnectionPot, rootModelDataPot.toPotView, ctxPot).tupled.toOption
-            .map: (wsConnection, rootModelData, ctx) =>
-              import ctx.given
+      .useResourceOnMount:
+        // Reconnect(WebSocketClient[IO]).connectHighLevel(WSRequest(EventWsUri))
```

Comment:
Commented line?

Reply:
Left on purpose, we will probably use this when it's implemented in http4s.